core/network/src/main/java/com/android/developers/androidify/ondevice/LocalSegmentationDataSource.kt

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2025 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.android.developers.androidify.ondevice
+
+import android.graphics.Bitmap
+import android.util.Log
+import com.google.android.gms.common.moduleinstall.InstallStatusListener
+import com.google.android.gms.common.moduleinstall.ModuleInstallClient
+import com.google.android.gms.common.moduleinstall.ModuleInstallRequest
+import com.google.android.gms.common.moduleinstall.ModuleInstallStatusUpdate
+import com.google.android.gms.common.moduleinstall.ModuleInstallStatusUpdate.InstallState.STATE_CANCELED
+import com.google.android.gms.common.moduleinstall.ModuleInstallStatusUpdate.InstallState.STATE_FAILED
+import com.google.mlkit.vision.common.InputImage
+import com.google.mlkit.vision.segmentation.subject.SubjectSegmentation
+import com.google.mlkit.vision.segmentation.subject.SubjectSegmenterOptions
+import kotlinx.coroutines.CancellableContinuation
+import kotlinx.coroutines.suspendCancellableCoroutine
+import javax.inject.Inject
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+
+interface LocalSegmentationDataSource {
+    suspend fun removeBackground(bitmap: Bitmap): Bitmap
+}
+
+class LocalSegmentationDataSourceImpl @Inject constructor(
+    private val moduleInstallClient: ModuleInstallClient
+) : LocalSegmentationDataSource {
+    private val segmenter by lazy {
```

Comment:
This class is used as a singleton right? `segmentor` has a close function if not that should probably be called if you are creating and destroying this class

Reply:
Its not closed off in the official sample for it - so i dont think we need to call it https://github.com/googlesamples/mlkit/blob/80d6bc64a0b88e18666f26feee0219995f1a86b2/android/vision-quickstart/app/src/main/java/com/google/mlkit/vision/demo/kotlin/subjectsegmenter/SubjectSegmenterProcessor.kt